packages/grafana-data/src/types/variables.ts

```diff
@@ -0,0 +1,17 @@
+import { OperatorFunction } from 'rxjs';
+
+import { DataQuery, DataQueryRequest, MetricFindValue } from './datasource';
+import { DataFrame } from './dataFrame';
+import { ScopedVars } from './ScopedVars';
+
+export interface VariableSupport<TQuery extends DataQuery = DataQuery> {
+  /**
+   * Defines how to transform the query provided from the variable into the DataQueryRequest used to query for MetricFindValues.
+   */
+  toDataQueryRequest: (query: any, scopedVars: ScopedVars) => DataQueryRequest<TQuery>;
```

Comment:
This looks a bit strange. The data source cannot define the full DataQueryRequest (it has time range, all sorts of information missing in this context). 

I was thinking more of something like 

toDataQuery(query: any) => TQuery

Reply:
I see, thank you that makes a lot of sense. 
What is your thoughts about Ryans concern about adding `CoreApp.Variables`? I'm struggling to solve the Prometheus migration without adding that or extending the PromQuery interface with a new prop.